File: application/src/main/java/de/pbauerochse/worklogviewer/domain/timerangeprovider/CustomTimerangeProvider.java
package de.pbauerochse.worklogviewer.domain.timerangeprovider;

import de.pbauerochse.worklogviewer.domain.ReportTimerange;
import de.pbauerochse.worklogviewer.report.TimeRange;

import java.time.LocalDate;

/**
 * TimerangeProvider for custom provided dates
 */
public class CustomTimerangeProvider extends BaseTimerangeProvider {

    private final TimeRange timeRange;

    CustomTimerangeProvider(LocalDate startDate, LocalDate endDate) {
        this.timeRange = new TimeRange(startDate, endDate);
    }

    @Override
    public TimeRange getTimeRange() {
        return timeRange;
    }

    @Override
    public ReportTimerange getReportTimerange() {
        return ReportTimerange.CUSTOM;
    }

}

File: application/src/main/java/de/pbauerochse/worklogviewer/timerange/CustomTimerangeProvider.kt
package de.pbauerochse.worklogviewer.timerange

import de.pbauerochse.worklogviewer.report.TimeRange
import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted
import java.time.LocalDate

/**
 * TimerangeProvider for a custom timerange where the start and end date is
 * provided by the user
 */
internal object CustomTimerangeProvider : TimerangeProvider {
    override val label: String = getFormatted("timerange.custom")
    override val settingsKey: String = "CUSTOM"
    override val isComputed: Boolean = false
    override fun buildTimeRange(start: LocalDate?, end: LocalDate?): TimeRange {
        checkNotNull(start) { getFormatted("timerange.custom.startrequired") }
        checkNotNull(end) { getFormatted("timerange.custom.endrequired") }
        return TimeRange(start, end)
    }
}
